### ISSUE TITLE
Greengrass component name case insensitivity

### ISSUE THREAD

**Issue by thair116** [aws-greengrass/aws-greengrass-nucleus#1194]
**Describe the bug**
Greengrass fails to save artifacts into `packages` directory when there is an existing component with the same, case-different name

**To Reproduce**
Deploy a component called `Component`
Deploy a second component called `component` that depends on an artifact (in our case, a config file pulled from s3) 

**Expected behavior**
`component` to successfully retrieve S3 and save it, OR an error to be raised about the name collision 

**Actual behavior**
Silent failure. The component comes up sometimes, and fails other times (with missing config file error). I _think_ it succeeds on initial come-up, but fails to restart thereafter (but cannot confirm that). I can can confirm that no `component` folder is made in `packages/artifacts`, and the `packages/artifacts/Component` folder still exists but is empty. 

deleting the `packages/artifacts/Component` folder and creating `packages/artifacts/component` fixed the issue
`
**Environment**
 - OS: macOS 11.4
 - JDK version: openjdk 17.0.2 2022-01-18
 - Nucleus version: 2.5.3

**Additional context**
mac

E.g. what is the impact of the bug?

**Comment by MikeDombo** [comment]
Hi Travis,
Thank you for submitting your issue. Greengrass's configuration is entirely case insensitive, which includes component names. Therefore, your component names must be unique when compared in a case insensitive way. Additionally, MacOS's file system is case insensitive by default which will also cause confusion.

**Comment by thair116** [comment]
Thanks for the reply @MikeDombo. 

> your component names must be unique when compared in a case insensitive way

If thats the expectation, there seems to be a discrepancy between the Greengrass UI in AWS Console (which is case-sensitive and will show `component` and `Component` as two different components) and greengrass nucleus, which will treat them as the same

**Comment by MikeDombo** [comment]
Thanks again for bringing this up. I'm passing it along to our cloud team to see if they can/will/want to enforce uniqueness case insensitively.

**Comment by thair116** [comment]
>  this would seem to be incredibly confusing to have meaningfully different components with the same name and only different capitalization

totally agree. :) This is not a use case that would be valuable, more just a case where catching the issue early and raising an error would be really helpful. We did not realize we had two components with the same name and different case (two engineers were working in parallel) and it took a while to connect our missing artifacts issue to the difference in case

**Comment by MikeDombo** [comment]
Thanks once again for the feedback. I have additionally reached out to have our documentation updated to reflect this.
As I mentioned earlier, our cloud team will also be looking into enforcing this from the cloud side.

Closing this issue now as a wontfix in the Nucleus. Feel free to comment and reopen if necessary.
